src/modules/tradeFormValidation/hooks/useTradeFormButtonContext.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react'
+
+import { useToggleWalletModal } from 'legacy/state/application/hooks'
+
+import { useWrapNativeFlow } from 'modules/trade'
+import { useDerivedTradeState } from 'modules/trade/hooks/useDerivedTradeState'
+import { useTradeQuote } from 'modules/tradeQuote'
+import { useWalletDetails } from 'modules/wallet'
+
+import { TradeFormButtonContext } from '../types'
+
+interface TradeFormCallbacks {
+  doTrade(): void
+  confirmTrade(): void
+}
+
+export function useTradeFormButtonContext(
+  defaultText: string,
+  callbacks: TradeFormCallbacks
+): TradeFormButtonContext | null {
+  const { state: derivedState } = useDerivedTradeState()
+  const wrapNativeFlow = useWrapNativeFlow()
+  const { isSupportedWallet } = useWalletDetails()
+  const quote = useTradeQuote()
+  const toggleWalletModal = useToggleWalletModal()
+
+  return useMemo(() => {
+    if (!derivedState) return null
+
+    return {
+      defaultText,
+      derivedState,
+      quote,
+      isSupportedWallet,
+      ...callbacks,
+      wrapNativeFlow() {
+        wrapNativeFlow()
+      },
```

Comment:
Same here as the previous PR, can you do without the extra fn?

```suggestion
      wrapNativeFlow,
```

Reply:
Fixed